core/src/main/java/org/geysermc/geyser/extension/GeyserExtensionLoader.java

```diff
@@ -157,16 +165,17 @@ protected void loadAllExtensions(@NonNull ExtensionManager extensionManager) {
                             return;
                         }
 
-                        // Completely different API version
-                        if (description.majorApiVersion() != Geyser.api().majorApiVersion()) {
-                            GeyserImpl.getInstance().getLogger().error(GeyserLocale.getLocaleStringLog("geyser.extensions.load.failed_api_version", name, description.apiVersion()));
-                            return;
-                        }
-
-                        // If the extension requires new API features, being backwards compatible
-                        if (description.minorApiVersion() > Geyser.api().minorApiVersion()) {
-                            GeyserImpl.getInstance().getLogger().error(GeyserLocale.getLocaleStringLog("geyser.extensions.load.failed_api_version", name, description.apiVersion()));
-                            return;
+                        // Completely different API version - or if the extension requires new API features, being backwards compatible
+                        if (!GeyserApi.api().geyserApiVersion().isCompatible(description.majorApiVersion(), description.minorApiVersion(), description.patchApiVersion())) {
```

Comment:
I think we should have more methods in ApiVersion.
We want to deny extensions that don't have the same major version.
I don't even think we need to log a warning for plugins using an api version that is still compatible (same major version).
But if we do, I would want to run an  `description.version().isOlderThan(geyserApiVersion())` or `geyserApiVersion().isNewerThan(description.version())`

Reply:
We would not issues a warning though - isCompatible returns only false when the requested api version is newer than the given (outdated Geyser, or some dev extension), or when the major version is completely different.
The warning only exists since we would be switching from specifying base api version -> geyser api version; as to not break backwards compatibility.

Reply:
Right, forgot what the switch meant for already existing extensions their yml file.
But still, as long as the major version is the same there can't be situations where the minor and patch versions matter.
So we only have to do:
```java
if (!GeyserApi.api().geyserApiVersion().major() != description.major()) {
    // workaround for the switch to Geyser API versioning
    if (GeyserApi.api().baseApiVersion().major() != 1) {
        // incompatible extension version
    }
}
```

Reply:
This would unfortunately allow extensions that rely on newer api that is present to be loaded; which would crash Geyser :/
It in theory should not happen if Geyser is up to date.. but we cant guarantee that.

Reply:
I understand what you're trying to say now.
The api version in the extension should be the minimum supported version, so we should indeed check if the api version is newer or equal to the minimum supported api version of the extension. However we only have to check the major and minor version if we follow semantic versioning.

Reply:
> However we only have to check the major and minor version if we follow semantic versioning

part of Redned's proposal on Discord was that api additions bump the patch version, which this PR was following. 

It would be nice to follow semantic versioning though. In that case, api additions would bump minor. Every major minecraft update (1.21, 1.22) would bump the major version, allowing for deprecations to be removed (in spirit with red's proposal). 

That would cause extensions to break on every minecraft update though. I vaguely remember a conversation about wanting to avoid that. I guess the api major could be bumped for removals, but not necessarily for every large minecraft update.

Reply:
wouldnt that also require Geyser to be bumped on each major version, since api is supposed to follow geyser versioning?

I personally would stick to what is proposed tbh; but that's just my two cents

Reply:
> I guess the api major could be bumped for removals, but not necessarily for every large minecraft update.
Yes, there is no need to do a major version bump for Minecraft updates because there have been no changes to the api